helios-client/src/main/java/com/spotify/helios/auth/AuthProvider.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2015 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.auth;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import com.spotify.helios.client.RequestDispatcher;
+
+public interface AuthProvider {
+
+  /**
+   * Get the current value to use in the "Authorization" header.
+   *
+   * @return {@code null} if the value is unavailable.
+   */
+  String currentAuthorization();
```

Comment:
It doesn't return a token per se -- it returns the value to be put in the `Authorization` header. Which typically includes things like the scheme-name.

`currentAuthorizationHeader()` ?